flink-table-store-common/src/main/java/org/apache/flink/table/store/codegen/Projection.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.codegen;
+
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.binary.BinaryRowData;
+
+/**
+ * Interface for code generated projection, which will map a RowData to another BinaryRowData.
+ * Copied from Flink.
+ */
+public interface Projection {
+
+    BinaryRowData apply(RowData row);
```

Comment:
Should `BinaryRowData` be used here? How about `RowData`?

Reply:
I found the class in Flink is `Projection<IN extends RowData, OUT extends RowData>`

Reply:
The project from the codegen must return `BinaryRowData`. We also need to use `BinaryRowData`. For example, in `BucketComputer`.